packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -192,6 +192,46 @@ export class SummarizerNode implements IRootSummarizerNode {
 		return { summary: result.summary, stats: result.stats };
 	}
 
+	public validateSummary() {
+		this.validateSummaryCore(false);
+	}
+
+	protected validateSummaryCore(parentSkipRecursion: boolean) {
+		assert(
+			this.wipSummaryLogger !== undefined,
+			0x1a3 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
+		);
+		assert(this.wipReferenceSequenceNumber !== undefined, 0x1a4 /* "Not tracking a summary" */);
+		let didNodeSummarize = this.wipLocalPaths !== undefined;
+
+		if (parentSkipRecursion) {
+			if (this._latestSummary !== undefined) {
```

Comment:
nit: Invert the if check so it's `if (undefined) { return } ...`

And please add a brief comment explaining the logic (maybe more succinct than what's in completeSummaryCore)